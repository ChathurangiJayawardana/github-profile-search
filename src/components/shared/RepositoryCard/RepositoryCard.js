import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import StarIcon from '../../../assets/images/star-icon.svg';
import colors from '../../../setup/colors';

const RepositoryCard = ({ title, description, stars }) => (
  <Container>
    <Title>{title}</Title>
    <Description>{description}</Description>
    <StarsContainer>
      <img src={StarIcon} alt="Repository Star Icon" />
      <Stars>{stars}</Stars>
    </StarsContainer>
  </Container>
);

const Container = styled.div`
  width: 750px;
  height: auto;
  background: transparent;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
`;

const Title = styled.p`
  font-family: 'Raleway';
  font-size: 35px;
  color: ${colors.purple}
  margin: 0;
`;

const Description = styled.p`
  font-family: Raleway;
  font-size: 20px;
  font-weight: 300;
  margin: 0;
  color: ${colors.black};
`;

const StarsContainer = styled.div`
  width: 300px;
  height: auto;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-top: 5px;
`;

const Stars = styled.span`
  font-family: Raleway;
  font-size: 20px;
  font-weight: 300;
  margin-left: 5px;
  color: ${colors.gray};
`;

RepositoryCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  stars: PropTypes.number,
};

RepositoryCard.defaultProps = {
  title: 'Lorem Pixel ðŸ˜…',
  description: 'Lorem Pixel Dolor asit emer.',
  stars: 0,
};

export default RepositoryCard;

