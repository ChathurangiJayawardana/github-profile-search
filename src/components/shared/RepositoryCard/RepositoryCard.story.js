import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import RepositoryCard from '.';

const stories = storiesOf('shared/RepositoryCard', module);

const props = {
  title: 'Lorem Pixel ðŸ˜…',
  description: 'Lorem Pixel Dolor asit emer.',
  stars: 0,
};

stories
  .add('default', () => <RepositoryCard {...props} />);


