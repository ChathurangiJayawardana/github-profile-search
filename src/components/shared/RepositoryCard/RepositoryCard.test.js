import React from 'react';
import { mount, shallow } from 'enzyme';

import RepositoryCard from '.';

const props = {
  title: 'Lorem Impsum',
  description: 'lorem impsum dolor a met.',
  stars: 123,
};

describe('RepositoryCard Component', () => {
  describe('renders', () => {
    it('renders without props', () => {
      const wrapper = shallow(<RepositoryCard />);
      expect(wrapper).toMatchSnapshot();
    });

    it('renders with props', () => {
      const rendered = shallow(<RepositoryCard {...props} />);
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('content', () => {
    it('should contain 2 paragraphs', () => {
      const wrapper = mount(<RepositoryCard />);

      expect(wrapper.find('p')).toHaveLength(2);
    });

    it('should contain 1 image', () => {
      const wrapper = mount(<RepositoryCard />);

      expect(wrapper.find('img')).toHaveLength(1);
    });

    it('should contain 1 span', () => {
      const wrapper = mount(<RepositoryCard />);

      expect(wrapper.find('span')).toHaveLength(1);
    });
  });
});

