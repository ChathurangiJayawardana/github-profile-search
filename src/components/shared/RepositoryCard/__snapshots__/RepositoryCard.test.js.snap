// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryCard Component renders renders with props 1`] = `ShallowWrapper {}`;

exports[`RepositoryCard Component renders renders without props 1`] = `ShallowWrapper {}`;
